### DISCUSSION TITLE
sentence reading word by word presentation with blank screen between words

### DISCUSSION THREAD

**Discussion by SRJongman** [jspsych/jsPsych#1074] [Q&A] [answered]
Hi,

I programmed a self-paced reading task where a sentence is shown word by word using the moving window plugin that Josh showed during the YouTube tutorial and this works well. However, participants are moving through the words too quickly. Is there a way to 1) add a blank screen between each word and, if possible, 2)have a minimum presentation duration of each word? Ideally I would use the moving window plugin - but perhaps I need to move to the html-keyboard-response plugin? However, there I don't know how to split my sentences into words - this is a different number of words for each sentence. And in my experiment each sentence is followed by a question so I can't just present word after blank screen after word etc. I hope this makes some sense! Any help would be appreciated, 

Suzanne

**Comment by becky-gilbert** [comment, marked as answer]
Hi @SRJongman! My approach would be to edit the moving window plugin so that it shows a blank screen between each word. After a response is made, instead of immediately calling the function that shows the next word, you could 
1) change to a blank display with `display_element.innerHTML = "";` and 
2) use `setTimeout` to call the function that shows the next word after a given delay. 

Sorry that I can't be more specific right now - I'm on my phone and don't have the moving window plugin code in front of me. But if you want to post your code here then we might be able to offer more specific advice.

**Reply by SRJongman** [reply]
Hi Becky,

Awesome, that worked! I was playing around with that before but didn't realize I could place the function to call the next word within setTimeout. Could you also help me with setting a minimum presentation duration for each word? I guess it also has to go into the function that handles a response but I am not sure how to handle it. This is the response function I have so far:

    // create a function for handling a response
    function after_response(response_info){
      display_element.innerHTML = "";
      rt.push(response_info.rt);
      jsPsych.pluginAPI.setTimeout(function(){
        current_word++;
        if(current_word == n_words){end_trial();} 
        else {show_stimulus(current_word);}
        }, 300);
    }

**Reply by becky-gilbert** [reply]
Hi @SRJongman, your code looks good to me! It looks like it should produce a 300 ms delay between the key press and the next word (or end of the trial). Is that not what happens? Or is that not what you were trying to do?

**Reply by SRJongman** [reply]
Hi Becky,
That was part one of what I wanted to achieve and yes, that works perfectly! Part two however is unrelated: I want the individual words to be shown for a minimum of 100 ms. So now if someone responds the word automatically disappears (and a blank screen follows of 300 ms). However, I want to always have a minimum presentation of 100 ms for the word. So if someone responds with a button press of 70 ms I want the word to stay on for another 30 ms before moving on to the blank screen. If someone responds >100 ms then the blank screen can be shown immediately. Is there an easy way to code that? 
Best,
Suzanne

**Reply by becky-gilbert** [reply]
Ah yes, I understand now, sorry. Ok, to do this I think you just need to check the value of the RT (`response_info.rt`) the `after_response` function. If the RT was long enough (>100 ms) then you'd change to the blank display and start the new 300 ms timer, before either showing the next word or ending the trial. If the RT was not long enough, then you'd set up another `setTimeout` timer that waits for the remaining amount of time, i.e. 100 ms - the RT. Because you'd be repeating some code here, I suggest creating a separate function that handles the switch to a blank screen. Then in the `after_response` function, you would either call the `switch_to_blank_screen` function immediately or after the remaining time has passed.

So you could try this:
```javascript
function switch_to_blank_screen() {
  display_element.innerHTML = "";
  jsPsych.pluginAPI.setTimeout(function(){
    current_word++;
    if (current_word == n_words) {
      end_trial(); 
    } else {
      show_stimulus(current_word);
    }
  }, 300);
}

function after_response(response_info){
  rt.push(response_info.rt);
  if (response_info.rt > 100) {
    // if enough time has passed, then switch to the blank screen now
    switch_to_blank_screen();
  } else {
    // if the RT is too short then wait the remaining amount of time before switching to the blank screen
    jsPsych.pluginAPI.setTimeout(function() {
      switch_to_blank_screen();
    }, 100 - response_info.rt);
  }
}
```

**Reply by SRJongman** [reply]
Hi Becky,

Thank you, that works wonderfully! Just a } missing before the second else. 

I really appreciate your help,

Suzanne

**Reply by becky-gilbert** [reply]
Fixed! Thanks for spotting that.
